internal/github/github_test.go

```diff
@@ -0,0 +1,874 @@
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if r.Method != http.MethodPost {
+					t.Errorf("expected POST request, got %s", r.Method)
+				}
+				if r.URL.Path != fmt.Sprintf("/repos/%s/%s/pulls", repo.Owner, repo.Name) {
+					t.Errorf("expected request to /repos/%s/%s/pulls, got %s", repo.Owner, repo.Name, r.URL.Path)
+				}
+
+				w.WriteHeader(tc.statusCode)
+				if tc.mockResponse != nil {
+					if err := json.NewEncoder(w).Encode(tc.mockResponse); err != nil {
+						t.Fatalf("failed to write mock response: %v", err)
+					}
+				}
+			})
+			client, server := newTestServerAndClient(t, handler)
+			defer server.Close()
+
+			pr, err := client.CreatePullRequest(context.Background(), repo, remoteBranch, title, tc.body)
+
+			if (err != nil) != tc.wantErr {
+				t.Errorf("CreatePullRequest() error = %v, wantErr %v", err, tc.wantErr)
+				return
+			}
+
+			if !tc.wantErr {
+				expectedPR := &PullRequestMetadata{
+					Repo:   repo,
+					Number: tc.mockResponse.GetNumber(),
+				}
+				if diff := cmp.Diff(expectedPR, pr); diff != "" {
+					t.Errorf("CreatePullRequest() mismatch (-want +got):\n%s", diff)
+				}
+			}
+		})
+	}
+}
+
+func TestGetCommit(t *testing.T) {
+	repo := &Repository{Owner: "test-owner", Name: "test-repo"}
+	sha := "abcdef123"
+	cases := []struct {
+		name         string
+		mockResponse *github.RepositoryCommit
+		statusCode   int
+		wantErr      bool
+	}{
+		{
+			name: "Successful get",
+			mockResponse: &github.RepositoryCommit{
+				SHA: github.Ptr("abcdef123"),
+			},
+			statusCode: http.StatusOK,
+			wantErr:    false,
+		},
+		{
+			name:       "GitHub API error",
+			statusCode: http.StatusInternalServerError,
+			wantErr:    true,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if r.Method != http.MethodGet {
+					t.Errorf("expected GET request, got %s", r.Method)
+				}
+				expectedPath := fmt.Sprintf("/repos/%s/%s/commits/%s", repo.Owner, repo.Name, sha)
+				if r.URL.Path != expectedPath {
+					t.Errorf("expected request to %s, got %s", expectedPath, r.URL.Path)
+				}
+
+				w.WriteHeader(tc.statusCode)
+				if tc.mockResponse != nil {
+					if err := json.NewEncoder(w).Encode(tc.mockResponse); err != nil {
+						t.Fatalf("failed to write mock response: %v", err)
+					}
+				}
+			})
+			client, server := newTestServerAndClient(t, handler)
+			defer server.Close()
+
+			commit, err := client.GetCommit(context.Background(), repo, sha)
+
+			if (err != nil) != tc.wantErr {
+				t.Errorf("GetCommit() error = %v, wantErr %v", err, tc.wantErr)
+				return
+			}
+
+			if !tc.wantErr {
+				if diff := cmp.Diff(tc.mockResponse, commit); diff != "" {
+					t.Errorf("GetCommit() mismatch (-want +got):\n%s", diff)
+				}
+			}
+		})
+	}
+}
+
+func TestGetRawContent(t *testing.T) {
```

Comment:
I believe similar tests also added in https://github.com/googleapis/librarian/pull/810/files.